inc/namespace.php

```diff
@@ -262,3 +241,97 @@ function autoenqueue( string $manifest_path, string $target_bundle, array $optio
+ *
+ * @param string $manifest_path File system path for an asset manifest JSON file.
+ * @param string $target_asset  Asset to retrieve within the specified manifest.
+ * @param array  $options {
+ *     @type string $handle       Handle to use when enqueuing the asset. Required.
+ *     @type array  $dependencies Script or Style dependencies. Optional.
+ * }
+ */
+function register_asset( string $manifest_path, string $target_asset, array $options = [] ) : void {
+	$defaults = [
+		'dependencies' => [],
+	];
+	$options = wp_parse_args( $options, $defaults );
+
+	$manifest_folder = trailingslashit( dirname( $manifest_path ) );
+
+	$asset_uri = Manifest\get_manifest_resource( $manifest_path, $target_asset );
+
+	// If we fail to match a .css asset, try again with .js in case there is a
+	// JS wrapper for that asset available (e.g. when using DevServer).
+	if ( empty( $asset_uri ) && is_css( $target_asset ) ) {
+		$asset_uri = Manifest\get_manifest_resource( $manifest_path, preg_replace( '/\.css$/', '.js', $target_asset ) );
+	}
+
+	// If asset is not present in manifest, attempt to resolve the $target_asset
+	// relative to the folder containing the manifest file.
+	if ( empty( $asset_uri ) && file_exists( $manifest_folder . $target_asset ) ) {
+		// TODO: Consider warning in the console if the asset could not be found.
+		// (Failure should be allowed for CSS files; they are not exported in dev).
+		$asset_uri = $target_asset;
+	}
+
+	// Use the asset URI as the asset handle if no handle was provided.
+	if ( empty( $options['handle'] ) ) {
+		$options['handle'] = $asset_uri;
+	}
+
+	// Reconcile static asset build paths relative to the manifest's directory.
+	if ( strpos( $asset_uri, '//' ) === false ) {
+		$asset_uri = Paths\get_file_uri( $manifest_folder . $asset_uri );
+	}
+
+	if ( is_css( $asset_uri ) ) {
+		wp_register_style(
+			$options['handle'],
+			$asset_uri,
+			$options['dependencies'],
+			$asset_uri
+		);
+	} else {
+		Admin\maybe_setup_ssl_cert_error_handling( $asset_uri );
+		wp_register_script(
+			$options['handle'],
+			$asset_uri,
+			$options['dependencies'],
+			$asset_uri,
+			true
+		);
+	}
+}
+
+/**
+ * Attempt to register and then enqueue a particular script bundle from a manifest.
+ *
+ * @param string $manifest_path File system path for an asset manifest JSON file.
+ * @param string $target_asset  Asset to retrieve within the specified manifest.
+ * @param array  $options {
+ *     @type string $handle       Handle to use when enqueuing the asset. Required.
+ *     @type array  $dependencies Script or Style dependencies. Optional.
+ * }
+ */
+function enqueue_asset( string $manifest_path, string $target_asset, array $options = [] ) : void {
+	register_asset( $manifest_path, $target_asset, $options );
+
+	// $target_asset will share a filename extension with the enqueued asset.
+	if ( is_css( $target_asset ) ) {
+		wp_enqueue_style( $options['handle'] );
```

Comment:
You automatically create a handle if none is present in the options in `register_asset`, but don't use that auto-generated handle here, so you could end up not passing a handle at, I think. What about returning the handle from `register_asset` every time and using that here?

Reply:
Good catch, fixed!